pkg/machine/provider/platform.go

```diff
@@ -49,19 +49,34 @@ func SupportedProviders() []define.VMType {
 
 // InstalledProviders returns the supported providers that are installed on the host
 func InstalledProviders() ([]define.VMType, error) {
-	cfg, err := config.Default()
+	installed, err := IsInstalled(define.QemuVirt)
 	if err != nil {
 		return nil, err
 	}
-	_, err = cfg.FindHelperBinary(qemu.QemuCommand, true)
-	if errors.Is(err, fs.ErrNotExist) {
-		return []define.VMType{}, nil
-	}
-	if err != nil {
-		return nil, err
+	if installed {
+		return []define.VMType{define.QemuVirt}, nil
 	}
+	return []define.VMType{}, nil
+}
```

Comment:
Wouldn't the change you made here have to get applied to `provider/platform_darwin.go` and `provider/platform_windows.go` as well?

Reply:
Updated to a unified function, thanks!